### PR TITLE
lnd_services: make unlock wait compatible with lnd 0.13

### PR DESCRIPTION
**Pull request by guggero**
The new version 0.13 of lnd doesn't return the same RPC error codes
anymore if its wallet is still locked. We need to do some string
matching here to avoid needing to pull in the 0.13 version as a
dependency which would nullify our efforts to have backward compatible
branches of lndclient.

#### Pull Request Checklist

- [X] PR is opened against correct version branch.
- [ ] Version compatibility matrix in the README and minimal required version
      in `lnd_services.go` are updated.
- [ ] Update `macaroon_recipes.go` if your PR adds a new method that is called
  differently than the RPC method it invokes.

### CODE REVIEW

**Review by carlaKC — APPROVED** [review]
LGTM! 

Just to confirm, the idea is to add this here so that wait for unlocked works for old + 0.13 clients, and then we will get rid of this on the 0.13 branch and just use the unlocker? 

Tested the following cases with `loopd` / v0.12 + v0.13 and all in order!
- Lnd not running at all at `loopd` startup, client shuts down
- Lnd locked at `loopd` startup, client waits for unlock
- Lnd unlocked at `loopd` startup, client ok

**Comment by guggero** [comment]
> Just to confirm, the idea is to add this here so that wait for unlocked works for old + 0.13 clients, and then we will get rid of this on the 0.13 branch and just use the unlocker?

Yes, the `lnd-13-0` branch already has its own unlocking logic that uses the new status RPC. When rebasing that branch I'm going to remove this logic again as we require `lnd v0.13.0` to run that branch anyway.

File: lnd_services.go [hunk 1]
```diff
@@ -448,6 +459,14 @@ func getLndInfo(ctx context.Context, ln lnrpc.LightningClient,
 		// Unlocked, ok: lnd is unlocked, no error
 		// Unlocked, not ok: lnd is unlocked but in bad state, err
 		//
+		// For nodes of version 0.13 and later, we don't get any of the
+		// Unimplemented/Unavailable codes anymore since the RPC server
+		// is up the whole time. Instead we need to look at the error
+		// message itself. We can't just use the state RPC in that case
+		// either because that would mean we'd need to pull in the 0.13
```
Comment (halseth) [inline lnd_services.go:466]: Good comment, now I understand the problem better 👍

**Review by halseth — APPROVED** [review]
Great, turned out not so bad 💯

**Comment by guggero** [comment]
Thanks for the reviews!
I exported the `IsUnlockError()` function so we can also use it in LiT. Did a smoke test and this seems to work fine in the combinations I tested.
Going to merge and rebase the other branches.

**Pull request merged**